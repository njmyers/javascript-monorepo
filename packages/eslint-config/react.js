module.exports = {
  parser: 'babel-eslint',
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
  env: {
    'shared-node-browser': true,
  },
  extends: ['plugin:react/recommended'],
  plugins: ['react', 'react-hooks', 'jsx-a11y'],
  rules: {
    'jsx-a11y/label-has-associated-control': 'error',
    'react/display-name': 'off',
    'react/prop-types': 'off',
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'error',
  },
  settings: {
    react: {
      version: 'detect',
    },
  },
};
