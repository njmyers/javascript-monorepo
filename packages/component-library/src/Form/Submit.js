// @flow
import * as React from 'react';
import FormElement from './FormElement';

const Submit = React.forwardRef((props, ref) => (
  <FormElement as="input" forwardedRef={ref} {...props} />
));

Submit.defaultProps = {
  name: 'submit',
  type: 'submit',
  value: 'submit',
  block: 'formInput',
  required: false,
  placeholder: '',
};

export default Submit;
