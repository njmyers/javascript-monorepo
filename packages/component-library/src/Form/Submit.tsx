// @ts-nocheck

import * as React from 'react';
import Element from './Element';

const Submit = React.forwardRef((props, ref) => (
  <Element as="input" forwardedRef={ref} {...props} />
));

Submit.defaultProps = {
  name: 'submit',
  type: 'submit',
  value: 'submit',
  block: 'formInput',
  required: false,
  placeholder: '',
};

export default Submit;
