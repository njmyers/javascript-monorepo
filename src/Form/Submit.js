// @flow
import * as React from 'react';
import BEM from '../BEM';

type Props = {
  name: string,
  type: string,
  value: string,
  block: string,
  modifiers?: string | Array<string> | void,
};

class Submit extends React.PureComponent<Props> {
  static defaultProps = {
    name: 'submit',
    type: 'submit',
    value: 'submit',
    block: 'formInput',
  };

  render() {
    return (
      <BEM block={this.props.block}>
        <div modifiers={this.props.modifiers}>
          <input
            element="input"
            modifiers={this.props.modifiers}
            type={this.props.type}
            value={this.props.value}
          />
        </div>
      </BEM>
    );
  }
}

export default Submit;
