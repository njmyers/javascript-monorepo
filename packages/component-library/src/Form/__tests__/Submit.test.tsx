import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Submit from '../Submit';

describe('it works', () => {
  test('it loads without crashing', () => {
    const render = TestRenderer.create(<Submit />);
    expect(render.toJSON()).toMatchSnapshot();
  });
});
