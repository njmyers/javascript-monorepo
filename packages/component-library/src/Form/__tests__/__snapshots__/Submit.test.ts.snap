// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it loads without crashing 1`] = `
<div
  className="formInput formInput"
>
  <label
    className="formInput_label formInput_label"
    htmlFor="submit"
  >
    submit
  </label>
  <input
    className="formInput_input formInput_input"
    name="submit"
    placeholder=""
    required={false}
    type="submit"
    value="submit"
  />
</div>
`;
